### PR TITLE
Fix a ton of potential crashers, odd code and redundant calls in ModelService

### PR DESCRIPTION
**Pull request by shana**
Ugh, what a mess. While fixing #1319, I came across a bunch of repository tests that were not actually using the code that the app is using.

Moved those tests to use the real API and deleted the old deprecated ModelService methods, and in the process fixed an old `TrackingCollection` bug where the `OriginalCompleted` observable would finish before all items have been processed.

Added exception catching to various places so we don't crash if keys aren't found or the network is down while accessing data.

This needs #1449 for tests to pass.

### CODE REVIEW

File: test/UnitTests/GitHub.App/Models/ModelServiceTests.cs [hunk 1]
```diff
@@ -157,20 +155,20 @@ public class TheGetAccountsMethod : TestBaseClass
...
             var fetched = await modelService.GetAccounts();
 
-            Assert.That(3, Is.EqualTo(fetched.Count));
-            Assert.That("snoopy", Is.EqualTo(fetched[0].Login));
-            Assert.That("github", Is.EqualTo(fetched[1].Login));
-            Assert.That("fake", Is.EqualTo(fetched[2].Login));
+            Assert.AreEqual(3, fetched.Count);
+            Assert.AreEqual("snoopy", fetched[0].Login);
+            Assert.AreEqual("github", fetched[1].Login);
+            Assert.AreEqual("fake", fetched[2].Login);
```
Comment (grokys) [inline test/UnitTests/GitHub.App/Models/ModelServiceTests.cs]: The `Assert.That` constraint model is preferred for nunit 3 apparently? See https://github.com/github/VisualStudio/pull/1411#issuecomment-358045651 and the surrounding comments for context on this.

**Comment by grokys** [comment]
Also, I think the fix for `TrackingCollection` should probably be in a separate PR...

File: test/UnitTests/GitHub.App/Models/ModelServiceTests.cs (hunk 1 above)
Comment (shana) [reply test/UnitTests/GitHub.App/Models/ModelServiceTests.cs]: Ah, ok. It's just that it inverts the order of the parameters, makes it even more confusing than usual. I'll fix these then.

Comment (grokys) [reply test/UnitTests/GitHub.App/Models/ModelServiceTests.cs]: Yeah, I think we need to switch the order of parameters in quite a few tests if we want to be technically correct. However, it doesn't actually impact the testing and it's tedious work...

**Pull request closed without merging**